app/controllers/Streamer.scala

```diff
@@ -187,7 +187,14 @@ final class Streamer(env: Env, apiC: => Api) extends LilaController(env):
       }
     }
 
-  private def AsStreamer(f: StreamerModel.WithUser => Fu[Result])(implicit ctx: Context) =
+  def subscribe(streamer: UserStr, set: Boolean) =
+    AuthBody { implicit ctx => me =>
+      if (set) env.relation.subs.subscribe(me.id, streamer.id)
+      else env.relation.subs.unsubscribe(me.id, streamer.id)
+      fuccess(Ok)
```

Comment:
make sure `Ok` is only returned after the future succeeds:

```scala
val result = if (set) ... else ...
result inject Ok // same as result.map(_ => Ok)
```

Reply:
Yes, I should also make sure a not-OK response results in an error for the user.